sections/06-vibrant-bodies.md

```diff
@@ -0,0 +1,201 @@
+
+I do not want to linger for long here on the technical details of _Flocking_, _Aconite_, or my software-based videos and compositions, since they are discussed extensively elsewhere<sup>7</sup>. However, I feel that it is necessary to briefly examine the mechanism by which these tools aspire to function as material vectors for further creative practices--my own and those of others.
+
+By virtue of the manner in which they are constructed and the nature of tools with which they are implicated, my software-based videos and music compositions were developed in a different way from most conventional software. Programming typically enacts a curious separation between code and data. Code represents the "logic" and "behaviour" of a piece of software. As Chun notes, source code is usually understood as the primary active agent of computation (19). In contrast, data is passive and raw; it serves as the material from which computational representations are formed. Programming languages are designed to process data, but are not themselves easily processable<sup>8</sup>. This means that ordinary programs are unable to understand or modify other programs<sup>9</sup>. Perhaps counter-intuitively, this oft-neglected computational self-referenceability, known as "homoiconicity"--the capacity for programs to treat themselves and other programs as data, and thus, to self-modify--is the prerequisite for individuals and communities (not just algorithms) to more directly participate in a convivial relationship with their software materials, bidirectionally as both users and creators (Basman et al 255).
+
+Programs made with Flocking and Aconite are data, not just code. This means, in particular, that they can be "understood" by programs and modified algorithmically as data, using the strategies that are typically used for other kinds of data processing. One particular technique is used extensively in Flocking programs to make them more amenable to material transformation: merging. In Flocking, instruments and signal processors are defined as "trees" of data that represent the connections amongst individual "unit generators," or signal producing/modulating algorithms. Even the compositional or "meta-level" material of a piece can be defined as a tree of signals, as in _Font Màgica<sup>10</sup>. By structuring a program as nested collections of named values, simple algorithms can be created (including the ones that are built directly into Flocking and its underlying support infrastructure) to "graft" or merge different signal trees together. A given instance of a signal tree created by one artist can be reconfigured without "forking" or changing the original source<sup>11</sup>, nor requiring there even to be an intent or active design to support such adaptation. Where free software legally mandates sharing, code-as-data participates in a form of implicit, integral gift economy where each work holds the potential to serve as the material for new ones.
```

Comment:
Comment from David: "I wonder about the phrase 'gift economy.’ Are you familiar with the Mauss stuff from which the term comes? I only ask because it’s not totally clear to me that what you mean jives with the term…you could easily re-word it if it’s not what you mean (or explain a bit if it is). "